### PR TITLE
46 pose 2 gps

### PR DESCRIPTION
**Pull request by patates-cipsi418**
implementation of nav2 navsat_transform to get gps position with the odometry.
remap /vectornav/imu to /imu for standardisation
remap /vectornav/gnss to /gps/fix for standardisation
add gps sensor to gazebo.

used tutorial: https://navigation.ros.org/tutorials/docs/navigation2_with_gps.html

### CODE REVIEW

File: src/rove_bringup/launch/vectornav.launch.py [hunk 1]
```diff
@@ -21,5 +21,9 @@ def generate_launch_description():
             executable='vn_sensor_msgs',
             output='screen',
             parameters=[vn_param_file],
+            remappings=[
+                ('/vectornav/gnss', '/gps/fix'),
```
Comment (SimonR99) [inline src/rove_bringup/launch/vectornav.launch.py:25]: pas fan de /gps/fix

File: src/rove_description/launch/amazon.launch.py [hunk 2]
```diff
@@ -92,22 +92,52 @@ def generate_launch_description():
...
        package='robot_localization',
        executable='ekf_node',
-       name='ekf_filter_node',
+       name='ekf_filter_node_local',
        output='screen',
        parameters=[os.path.join(pkg_rove_slam, 'config/ekf.yaml'),
                    {'use_sim_time': True}]
                    )
 
+    robot_localization_node_global = Node(
```
Comment (SimonR99) [inline src/rove_description/launch/amazon.launch.py:104]: add source for local + global

File: src/rove_slam/config/ekf.yaml [hunk 3]
```diff
@@ -24,3 +24,48 @@ ekf_filter_node:
...
+                       
+        imu0: /imu
+        imu0_config: [false, false, false,
+                      true,  true,  true,
+                      false, false, false,
+                      false, false, false,
+                      false, false, false]
+
+        odom1: /odometry/gps
+        odom1_config: [true,  true,  false,
```
Comment (SimonR99) [inline src/rove_slam/config/ekf.yaml:55]: vérifier les paramètres du gps (yam pitch roll, x, y, z, ...)

File: src/rove_slam/config/ekf.yaml [hunk 4]
```diff
@@ -24,3 +24,48 @@ ekf_filter_node:
...
+                       false, false, false,
+                       false, false, false,
+                       false, false, false,
+                       false, false, false]
+
+navsat_transform:
```
Comment (SimonR99) [inline src/rove_slam/config/ekf.yaml:61]: I don't think this should be there (gps config and not ekf). But arguable

**Review by SimonR99 — CHANGES_REQUESTED** [review]
make sure to understand difference between local and global

**Comment by patates-cipsi418** [comment]
La différence entre le local et le global est que le dernier possède les données GPS et comment le déplacement est sur la carte. Le local est smooth mais a du drift alors que le global aura une bonne position mais le mouvement est un step.

src: https://answers.ros.org/question/318328/robot_localization-dual-ekfhow-the-two-ekf-nodes-work-together/

File: src/rove_bringup/launch/vectornav.launch.py (hunk 1 above)
Comment (patates-cipsi418) [reply src/rove_bringup/launch/vectornav.launch.py:25]: can use /gps

File: src/rove_slam/config/ekf.yaml (hunk 4 above)
Comment (patates-cipsi418) [reply src/rove_slam/config/ekf.yaml:61]: both need to be used with each other, but can be seperated

File: src/rove_description/launch/amazon.launch.py [hunk 5]
```diff
@@ -92,22 +92,54 @@ def generate_launch_description():
         }.items(),
     )
 
-    robot_localization_node = Node(
+    # used tutorial: https://navigation.ros.org/tutorials/docs/navigation2_with_gps.html
+
```
Comment (SimonR99) [inline src/rove_description/launch/amazon.launch.py:96]: remove space

**Review by SimonR99 — APPROVED** [review]
lgtm

**Pull request merged**